[P0][Bug] Duplicate `Error` discriminant `39` breaks the build

**Difficulty:** Beginner

## Context / Problem
`cargo check` fails on `main` with `error[E0081]: discriminant value '39' assigned more than once`. Both `Error::InvalidPlatformFee` and `Error::TransferAlreadyPending` are declared with `= 39` in `src/errors.rs:85-87`. The crate does not compile.

```rust
// src/errors.rs
InvalidPlatformFee = 39,
TransferAlreadyPending = 39,   // <- E0081
```

Reproduction:
```bash
cargo check
# error[E0081]: discriminant value `39` assigned more than once
```

## Expected Behavior
`cargo check` and `cargo build --target wasm32-unknown-unknown --release` succeed on `main`.

## Proposed Solution / Implementation Details
In `src/errors.rs`, change `TransferAlreadyPending = 39` to `TransferAlreadyPending = 40`. Verify no other variant collides (grep the file for the new value). Search the test suite for any hard-coded `Error` integer code that may depend on the discriminant — string/symbolic error matching is fine, integer matching needs updating.

## Acceptance Criteria
- [ ] `cargo check` exits 0 on `main`.
- [ ] `cargo build --target wasm32-unknown-unknown --release` produces a WASM artifact.
- [ ] A test explicitly asserts `Error::TransferAlreadyPending` is returned by `initiate_campaign_transfer` when a transfer is already pending (must exist or be added).
- [ ] CI passes after this change in isolation.

---
_Filed as part of the external audit captured in [ROADMAP.md](#) (held locally by maintainer). Companion P0 issues file this together with #2–#5._

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -7,6 +7,28 @@ on:
     branches: [main]
 
 jobs:
+  # Issue #403: enforce formatting on every PR.
+  fmt:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
+        with:
+          components: rustfmt
+      - run: cargo fmt --all -- --check
+
+  # Issue #403: enforce clippy with -D warnings so unreachable_code and
+  # similar lints (see #401) cannot land silently again.
+  clippy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
+        with:
+          components: clippy
+          targets: wasm32-unknown-unknown
+      - run: cargo clippy --all-targets --features testutils -- -D warnings
+
   test:
     runs-on: ubuntu-latest
     steps:

--- a/src/errors.rs
+++ b/src/errors.rs
@@ -84,5 +84,5 @@ pub enum Error {
     /// The provided platform fee exceeds the maximum allowed basis points.
     InvalidPlatformFee = 39,
     /// A campaign transfer is already pending; cancel it before initiating a new one.
-    TransferAlreadyPending = 39,
+    TransferAlreadyPending = 40,
 }

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -147,7 +147,7 @@ impl ProofOfHeart {
         // a silently-clamped value. Matches the validation contract in
         // `update_platform_fee` and `set_campaign_fee_override`.
         if platform_fee > PLATFORM_FEE_MAX_BPS {
-            return Err(Error::ValidationFailed);
+            return Err(Error::InvalidPlatformFee);
         }
 
         // Validate that the address is a real SEP-41 token contract by probing
@@ -160,10 +160,6 @@ impl ProofOfHeart {
         )
         .map_err(|_| Error::InvalidTokenContract)?
         .map_err(|_| Error::InvalidTokenContract)?;
-
-        if platform_fee > PLATFORM_FEE_MAX_BPS {
-            return Err(Error::InvalidPlatformFee);
-        }
 
         bump_instance_ttl(&env);
         set_admin(&env, &admin);
@@ -1376,7 +1372,6 @@ impl ProofOfHeart {
         Self::require_not_paused(&env)?;
         if new_fee > PLATFORM_FEE_MAX_BPS {
             return Err(Error::InvalidPlatformFee);
-            return Err(Error::ValidationFailed);
         }
         let old_fee = get_platform_fee(&env);
         bump_instance_ttl(&env);

--- a/src/test.rs
+++ b/src/test.rs
@@ -106,9 +106,10 @@ fn test_platform_fee_cap_enforcement() {
     let contract_id = env.register_contract(None, ProofOfHeart);
     let client = ProofOfHeartClient::new(&env, &contract_id);
 
-    // Issue #343: init rejects fees above the cap rather than silently clamping.
+    // Issue #343 / #402: init rejects fees above the cap with InvalidPlatformFee
+    // (matching the CHANGELOG and update_platform_fee's contract).
     let res = client.try_init(&admin, &token_address, &5000);
-    assert_eq!(res.unwrap_err().unwrap(), Error::ValidationFailed);
+    assert_eq!(res.unwrap_err().unwrap(), Error::InvalidPlatformFee);
 
     // Re-init with the maximum allowed fee and verify it applies end-to-end.
     client.init(&admin, &token_address, &1000);
@@ -1057,10 +1058,11 @@ fn test_update_platform_fee() {
     assert_eq!(data_vec.get(0).unwrap(), 300);
     assert_eq!(data_vec.get(1).unwrap(), 500);
 
-    // Issue #343: fees above the cap are rejected, not silently clamped.
+    // Issue #343 / #401: fees above the cap are rejected with InvalidPlatformFee.
+    // (The earlier duplicate ValidationFailed assertion was unreachable because
+    // update_platform_fee only ever returned InvalidPlatformFee — see #401.)
     let result = client.try_update_platform_fee(&5000);
     assert_eq!(result.unwrap_err().unwrap(), Error::InvalidPlatformFee);
-    assert_eq!(result.unwrap_err().unwrap(), Error::ValidationFailed);
 }
 
 #[test]

